olive/engine/engine.py

```diff
@@ -710,19 +734,21 @@ def _cache_run(self, pass_name: int, pass_config: dict, input_model_id: str, out
             "output_model_id": output_model_id,
         }
         input_model_number = input_model_id.split("_")[0]
-        run_json_path = self.get_run_json_path(pass_name, input_model_number, pass_config)
+        run_json_path = self.get_run_json_path(pass_name, input_model_number, pass_config, accelerator_spec)
         try:
             with open(run_json_path, "w") as f:
                 json.dump(run_json, f, indent=4)
         except Exception as e:
             logger.error(f"Failed to cache run: {e}")
 
-    def _load_run(self, input_model_id: str, pass_name: int, pass_config: dict):
+    def _load_run(self, input_model_id: str, pass_name: int, pass_config: dict, accelerator_spec: AcceleratorSpec):
         """
         Load the run from the cache directory.
         """
         input_model_number = input_model_id.split("_")[0]
-        run_json_path = self.get_run_json_path(pass_name, input_model_number, pass_config)
+        run_json_path = self.get_run_json_path(pass_name, input_model_number, pass_config, accelerator_spec)
```

Comment:
Actually, I think this logic might not work. 
For example, we ran with accelerator_spec `A` which returns `is_accelerator_agnostic=True` and then `B` which returns `False`. The second run for `B` will reuse the run for `A` even though that is not correct. 

Since we now have a pass method for `is_accelerator_agnostic`, we can directly provide the correct value for `accelerator_spec` for this method at line `823`